File: frontend/src/types/account.ts
export type AccountStatus = "new" | "warming" | "active" | "cooldown" | "blocked" | "verified";

export interface Account {
  id: number;
  user_id: number;
  owner_id: number;
  telegram_id: number;
  phone: string | null;
  username: string | null;
  first_name: string | null;
  status: AccountStatus;
  proxy_id: number | null;
  device_config: Record<string, unknown> | null;
  last_device_regenerated_at?: string | null;
  warming_actions_completed: number;
  target_warming_actions: number;
  warming_started_at: string | null;
  last_activity_at: string | null;
  cooldown_until: string | null;
  created_at: string;
  updated_at: string;
}

export interface AccountVerifyResponse {
  needs_password: boolean;
  account: Account | null;
}

File: backend/app/schemas/account.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field

from app.schemas.sanitization import SanitizedModel


class AccountStatus(str, Enum):
    new = "new"
    warming = "warming"
    active = "active"
    cooldown = "cooldown"
    blocked = "blocked"
    verified = "verified"


class AccountCreate(SanitizedModel):
    telegram_id: int
    user_id: int | None = None
    phone: str | None = Field(default=None, max_length=32)
    username: str | None = Field(default=None, max_length=255)
    first_name: str | None = Field(default=None, max_length=255)
    status: AccountStatus | None = None
    proxy_id: int | None = None
    device_config: dict | None = None


class AccountUpdate(SanitizedModel):
    phone: str | None = Field(default=None, max_length=32)
    username: str | None = Field(default=None, max_length=255)
    first_name: str | None = Field(default=None, max_length=255)
    status: AccountStatus | None = None
    proxy_id: int | None = None
    device_config: dict | None = None
    warming_started_at: datetime | None = None
    last_activity_at: datetime | None = None
    warming_actions_completed: int | None = Field(default=None, ge=0)
    target_warming_actions: int | None = Field(default=None, ge=0)
    cooldown_until: datetime | None = None
    last_device_regenerated_at: datetime | None = None


class AccountResponse(BaseModel):
    id: int
    user_id: int
    owner_id: int
    telegram_id: int
    phone: str | None
    username: str | None
    first_name: str | None
    status: AccountStatus
    proxy_id: int | None
    device_config: dict | None
    warming_actions_completed: int
    target_warming_actions: int
    warming_started_at: datetime | None
    last_activity_at: datetime | None
    cooldown_until: datetime | None
    last_device_regenerated_at: datetime | None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class AccountVerifyResponse(BaseModel):
    needs_password: bool = False
    account: AccountResponse | None = None
